src/microbe_stage/MicrobeAI.cs

```diff
@@ -119,249 +69,44 @@ public void Think(float delta, Random random, MicrobeAICommonData data)
-                if (predator != null)
-                {
-                    DealWithPredators(random);
-                }
-                else
-                {
-                    if (RollCheck(SpeciesActivity, 400, random))
-                    {
-                        lifeState = LifeState.PLANTLIKE_STATE;
-                        boredom = 0;
-                    }
-                    else
-                    {
-                        lifeState = LifeState.NEUTRAL_STATE;
-                    }
-                }
-
-                break;
-            }
-
-            case LifeState.PREDATING_STATE:
-            {
-                // Peg your prey
-                if (!preyPegged)
-                {
-                    prey = null;
-                    prey = GetNearestPreyItem(data.AllMicrobes);
-                    if (prey != null)
-                    {
-                        preyPegged = true;
-                    }
-                }
-
-                if (preyPegged && prey != null)
-                {
-                    DealWithPrey(data.AllMicrobes, random);
-                }
-                else
-                {
-                    if (RollCheck(SpeciesActivity, 400, random))
-                    {
-                        lifeState = LifeState.PLANTLIKE_STATE;
-                        boredom = 0;
-                    }
-                    else
-                    {
-                        lifeState = LifeState.NEUTRAL_STATE;
-                    }
-                }
-
-                break;
-            }
-
-            case LifeState.SCAVENGING_STATE:
-            {
-                if (targetChunk == null)
-                {
-                    targetChunk = GetNearestChunkItem(data.AllChunks);
-                }
-
-                if (targetChunk != null)
-                {
-                    DealWithChunks(targetChunk, data.AllChunks);
-                }
-                else
-                {
-                    if (!RollCheck(SpeciesOpportunism, 400, random))
-                    {
-                        lifeState = LifeState.NEUTRAL_STATE;
-                        boredom = 0;
-                    }
-                    else
-                    {
-                        lifeState = LifeState.SCAVENGING_STATE;
-                    }
-                }
-
-                break;
-            }
+            RunAndTumble(random);
         }
 
-        // Run reflexes
-        DoReflexes();
-
         // Clear the absorbed compounds for run and rumble
         microbe.TotalAbsorbedCompounds.Clear();
     }
 
-    // There are cases when we want either ||, so here's two state rolls
-    private static bool RollCheck(float ourStat, float dc, Random random)
-    {
-        return random.Next(0.0f, dc) <= ourStat;
-    }
-
-    private static bool RollReverseCheck(float ourStat, float dc, Random random)
-    {
-        return ourStat <= random.Next(0.0f, dc);
-    }
-
-    private void DoReflexes()
```

Comment:
Thsi was to make cells run away if someone starts engulfing them, do they still do this?

Reply:
In almost all cases a microbe large enough and close enough to engulf would meet the criteria to trigger PreyFlee behavior: https://github.com/Revolutionary-Games/Thrive/pull/2230/files#diff-773530b4c6ee33a2e02f7e74885912ed139c317feb8a48ccb8f61383ec0c4e4dR231

Upon second look though it would be possible for a very brave cell to ignore being engulfed, so I've added one more if check on the main function.